src/main/java/Parser.java

```diff
@@ -0,0 +1,71 @@
+import java.io.IOException;
+
+public class Parser {
+    private AddCommand add;
+    private DeleteCommand delete;
+    private MarkCommand mark;
+    private UnMarkCommand unmark;
+    private Ui ui;
+    private Storage stor;
+    private TaskList arrayLL;
+    private int type;
+    public Parser(Ui ui, Storage stor, TaskList arrayLL) {
+        this.ui = ui;
+        this.stor = stor;
+        this.arrayLL = arrayLL;
+    }
+
+    public void checkText(String userText) throws NeoException, IOException {
+
+        if (userText.equals("list") || userText.equals("List")) {
+            stor.Retrievedata();
+        }
+        String arr[];
+        arr = userText.split(" ", 2);
+        if (arr.length>1 && arr[0].equals("mark")){
+            this.mark = new MarkCommand(ui, stor, arrayLL);
+            mark.complete(arr[1]);
+        }
+        if (arr.length>1 && arr[0].equals("unmark")){
+            this.unmark = new UnMarkCommand(ui, stor, arrayLL);
+            unmark.complete(arr[1]);
+        }
+        if (arr.length>1 && arr[0].equals("delete")){
+            this.delete = new DeleteCommand(ui, stor, arrayLL);
+            delete.complete(arr[1]);
+        }
+        else {
+            try {
+                if (!userText.equals("list") && !userText.equals("List") && !arr[0].equals("unmark") && !arr[0].equals("mark") && !arr[0].equals("delete")) {
+                    if (arr.length > 1 && arr[0].equals("deadline")) {
+                        type = 0;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+                    }
+                    if (arr.length > 1 && arr[0].equals("event")) {
+                        type = 1;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+
+                    }
+                    if (arr.length > 1 && arr[0].equals("todo")) {
+                        type = 2;
+                        String tempi = arr[1];
+                        this.add = new AddCommand(ui, stor, arrayLL, type);
+                        add.complete(tempi);
+                    }
+                    if (arr.length == 1 && arr[0].equals("todo")) {
+                        throw new NeoException("sorry todo cannot be empty");
+                    }
+                    if (!arr[0].equals("todo") && !arr[0].equals("event") && !arr[0].equals("deadline")) {
+                        throw new NeoException("Sorry I don't know what that means");
+                    }
+                }
+            } catch (NeoException | IOException e) {
+                System.out.println(e.getMessage());
+            }
+        }
```

Comment:
This section of text could be simplified much more with a 'switch' statement instead of 'if-else' statement
For example:
```java
if (arr.length < 0) {
    throw new NeoException("Sorry I don't know what that means");
} else {
    String command = arr[0];
    switch (command) {
    case "mark": 
        this.mark = new MarkCommand(ui, stor, arrayLL);
        mark.complete(arr[1]);
        break;
    case "unmark":
        this.unmark = new UnMarkCommand(ui, stor, arrayLL);
        unmark.complete(arr[1]);
        break;
    case "delete":
        this.delete = new DeleteCommand(ui, stor, arrayLL);
        delete.complete(arr[1]);
        break;
    case "deadline": 
        //add the other cases and code in
        //below until the default case
    default:
        //throw NeoException if the command string does not match any of the cases
        throw new NeoException("Sorry I don't know what that means");
}
```
read up on the switch statement documentation for more clarifications.

Reply:
Understood, Will modify it 
Thank you